scripts/http-jsonp-detection.nse

```diff
@@ -0,0 +1,174 @@
+local callback_url = function(host, port, target)
+  local path, response, report
+  path = target .. "?callback=testing"
+  response = http.get(host, port, path)
+  if response and response.body and response.status==200 then
+
+    local status, func
+    status, func = checkjson(response.body)
+
+    if status == true then
+      if func == "testing" then
+        report = "Callback function is completely controllable from the URL"
+      else
+        local p = string.find(func1, "testing")
+        if p then 
+          report = "Callback function is partially controllable from URL"
+        end
+      end
+    end
+  end
+  return report
+end
+
+--The function tries to bruteforce through the most common callback variable
+local callback_bruteforce = function(host, port, target)
+  local response, path, report
+  for _,p in ipairs(callbacks) do 
+    path = target
+    path = path .. "?" .. p .. "=test"
+    response = http.get(host, port, path)
+    if response and response.body and response.status==200 then 
+
+      local status, func
+      status, func = checkjson(response.body)
+
+      if status == true then  
+        report = "JSONP endpoint found at " .. path .. ". Callback variable is " .. p
+        break
+      end
+    end
+  end 
+  return report
+end 
+
+action = function(host, port)
+  local path = stdnse.get_script_args(SCRIPT_NAME .. ".path") or "/"
+  local output = stdnse.output_table()
+  output = {}
+
+  -- crawl to find jsonp endpoints urls
+  local crawler = httpspider.Crawler:new(host, port, path, {scriptname = SCRIPT_NAME})
+
+  if (not(crawler)) then
+    return
+  end
+
+  crawler:set_timeout(10000)
+
+  while(true) do
+    local status, r = crawler:crawl()
+    if (not(status)) then
+      if (r.err) then
+        return stdnse.format_output(false, r.reason)
+      else 
+        break
+      end
+    end
+
+    local target = tostring(r.url)
+    target = url.parse(target)
+    target = target.path 
+
+    -- First we try to get the response and look for jsonp endpoint there 
+    if r.response and r.response.body and r.response.status==200 then
+
+      local report, status, func
+      status, func = checkjson(r.response.body)
+
+      if status == true then
+        --We have found JSONP endpoint
+        --Put it inside a returnable table.
+        report = "JSONP endpoint found at " .. target .. ". Function name is " .. func
+        table.insert(output, report)
+
+        --Try if the callback function is controllable from URL.     
+        report = callback_url(host, port, target)        
+        if report ~= nil then
+          table.insert(output,report)
+        end
+
+      else 
+
+        --Try to bruteforce through most comman callback URLs
+        report = callback_bruteforce(host, port, target)
+        if report ~= nil then
+          table.insert(output,report)
+        end
+      end 
+
+    end 
+
+  end
+
+  --A way to print returnable 
+  if next(output) then
+    return output
```

Comment:
Update the output so the XML tree returns something like:
```
<table key='jsonp_endpoints'>
<elem>URL1</elem>
</table>
```
While having the normal output return additional information like if its controllable or partial only. For normal output something like this works:

```
The following JSONP endpoints were detected:
URL1
URL2 (partially controllable)
```